Makefile

```diff
@@ -252,7 +252,7 @@ ifneq (,$(findstring CYGWIN,$(CONFIG_TARGET_OS)))
   SOURCE += os/windows/cpu-affinity.c os/windows/posix.c
   WINDOWS_OBJS = os/windows/cpu-affinity.o os/windows/posix.o lib/hweight.o
   LIBS	 += -lpthread -lpsapi -lws2_32 -lssp
-  FIO_CFLAGS += -DPSAPI_VERSION=1 -Ios/windows/posix/include -Wno-format
```

Comment:
It is not clear to me why this patch has been marked as "do not merge"? I like this change.

Reply:
@bvanassche :-)

You've caught me mid-test! I made that patch to be able to check for more type conversion issues but it's so ugly and split across two headers (one of which can sometimes come after the other).

Keeping `-Wno-format` was a lot smaller/simpler although its presence led to needing a lot of the other fixes you see in the rest of this patch series...

Reply:
For now I'll remove all the "do not merge" patches but I'll think about submitting the "enable format checking on Windows" in a separate pull request.